win32ss/user/ntuser/nonclient.c

```diff
@@ -511,7 +511,44 @@ DefWndDoSizeMove(PWND pwnd, WORD wParam)
 	      RECT newRect = unmodRect;
 
 	      if (!iconic && !DragFullWindows) UserDrawMovingFrame( hdc, &sizingRect, thickframe );
-	      if (hittest == HTCAPTION) RECTL_vOffsetRect( &newRect, dx, dy );
+          if (hittest == HTCAPTION)
+          {
+              // Restore window size if it is snapped
+              if ((pwnd->InternalPos.NormalRect.left != 0 && pwnd->InternalPos.NormalRect.top != 0 &&
+                   pwnd->InternalPos.NormalRect.right != 0 && pwnd->InternalPos.NormalRect.bottom != 0) &&
```

Comment:
What is this test for? To check whether NormalRect is an **empty rectangle**, or else?

Reply:
yes, replaced to RECTL_bIsEmptyRect